tox.ini

```diff
@@ -3,19 +3,13 @@ max-line-length = 100
 
 [tox]
 skipsdist = True
-envlist = lint,unit
+envlist = lint,unit,integration
 
 [testenv]
 basepython = python3
 whitelist_externals = tox
 passenv = HOME
-
-[testenv:lint]
-commands =
-    tox -c {toxinidir}/operators/argo-controller -e lint
-    tox -c {toxinidir}/operators/argo-server -e lint
-
-[testenv:unit]
+ignore_errors=true
 commands =
-    tox -c {toxinidir}/operators/argo-controller -e unit
-    tox -c {toxinidir}/operators/argo-server -e unit
+    tox -c {toxinidir}/operators/argo-controller -e {envname}
```

Comment:
Is there some way we can nicely parameterize both charm and test type? I'd like to able to run `tox -e foo-unit` when working on a single charm. [See here](https://github.com/canonical/istio-operators/blob/master/tox.ini) for where we use the opposite approach from this file (parameterized by charm instead of parameterized by test type).

This requires a little bit more manually specifying things, but what do you think about this?

```ini
[tox]
skipsdist = True
envlist = {foo,bar}-{unit,lint}

[testenv]
whitelist_externals = tox
changedir =
  foo: operators/foo
  bar: operators/bar
commands =
  unit: tox -e unit
  lint: tox -e lint
```

We can also run all unit testing at once with `tox -e {foo,bar}-unit`.

Reply:
I wanted exactly what you suggest and there feels like there should be a clean way to do it, but I couldn't figure out the syntax.  The trouble I ran into was that I needed the separate parts of the envname for different things, like:

```
commands =
    tox -c {toxinidir}/operators/{charm_name} -e {test_type}
    tox -c {toxinidir}/operators/{charm_name} -e {test_type}
```

do you know how to do this?  I searched a bit but couldn't find a good example.  I could add a command to parse the `envname` (I think?  I assume all commands are run in the same shell?) but that feels hacky.

Reply:
Refactored based on some suggestions from @knkski and it looks much better now